provisioner/utils/setup-env-openeuler.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```

Comment:
I hava another comments on this scripts.
openEuler is the OS deriving from CentOS like fedora, Rocky linux does. 
And this scricpt is almost same as [setup-env-centos.sh](https://github.com/apache/bigtop/blob/master/provisioner/utils/setup-env-centos.sh).
IMO, it's not necessary to add this openEuler script and please use `setup-env-centos.sh` for openEuler just like Fedora, rockylinux does.
Please refer to : https://github.com/apache/bigtop/blob/master/provisioner/docker/docker-hadoop.sh#L391:
`*-centos-*|*-fedora-*|*-opensuse-*|*-rockylinux-*|*-openeuler-*)  `

Reply:
Please also modify `openeuler` to `centos` in field of `distro:` in `config_openeuler-22.03.yaml`:
`distro: openeuler` --> `distro: centos`

Thanks. @MacChen01

Reply:
> I hava another comments on this scripts. openEuler is the OS deriving from CentOS like fedora, Rocky linux does. And this scricpt is almost same as [setup-env-centos.sh](https://github.com/apache/bigtop/blob/master/provisioner/utils/setup-env-centos.sh). IMO, it's not necessary to add this openEuler script and please use `setup-env-centos.sh` for openEuler just like Fedora, rockylinux does. Please refer to : https://github.com/apache/bigtop/blob/master/provisioner/docker/docker-hadoop.sh#L391: `*-centos-*|*-fedora-*|*-opensuse-*|*-rockylinux-*|*-openeuler-*) `

The code like this, which i modfiy setup-env-centos.sh file to support the openeuler os.
Using the dnf tool instead of yum tool to install package, the code on openeuler and centos env has been verfied.
please review it , i will push it if you agree it.

```
**if [ -f /etc/os-release ]; then
  . /etc/os-release
fi**

**if [ "${ID}" = "openEuler" ];then
  dnf install rng-tools 'dnf-command(config-manager)' -y 
else
  yum -y install rng-tools yum-priorities
fi**

if [ -x /usr/bin/systemctl ] ; then
    sed -i 's@ExecStart=/sbin/rngd -f@ExecStart=/sbin/rngd -f -r /dev/urandom@' /usr/lib/systemd/system/rngd.service
    systemctl daemon-reload
    systemctl start rngd
else
    sed -i.bak 's/EXTRAOPTIONS=\"\"/EXTRAOPTIONS=\"-r \/dev\/urandom\"/' /etc/sysconfig/rngd
    service rngd start
fi

if [ $enable_local_repo == "true" ]; then
    echo "Enabling local yum."
    yum -y install yum-utils

    case ${ID} in
        fedora | **openEuler**)
            sed -i 's/gpgcheck=1/gpgcheck=0/g' /etc/dnf/dnf.conf
            ;;
        centos)
            sudo echo "gpgcheck=0" >> /etc/yum.conf
            ;;
    esac

    if [ "${ID}" = "openEuler" ];then
      sudo dnf config-manager --add-repo file:///bigtop-home/output
    else
      sudo yum-config-manager --add-repo file:///bigtop-home/output
    fi
    sudo echo "gpgcheck=0" >> /etc/yum.repos.d/bigtop-home_output.repo
    sudo echo "priority=9" >> /etc/yum.repos.d/bigtop-home_output.repo
else
    echo "local yum = $enable_local_repo ; NOT Enabling local yum.  Packages will be pulled from remote..."
fi
```

Reply:
In my understanding, openEuler is similar with Fedora.
OpenEuler may also supports Yum.
May I aks why do you use the dnf tool instead of yum tool to install package?

Reply:
Just install `yum-utils`  on OpenEuler to support Yum tool.
https://github.com/apache/bigtop/blob/openEuler-support/provisioner/utils/setup-env-centos.sh#L38

Reply:
The yum-utils, yum-priorities and yum-config-manager,etc.  not found in openeuler OS, so we use the dnf tool.
![image](https://github.com/apache/bigtop/assets/11262916/fdb22966-23d2-4ba4-8446-ce15d47d20ba)

Reply:
OK, that makes sense. Please add comments to [setup-env-centos.sh](http://setup-env-centos.sh/) explaining why it was adjusted for openEuler:
```
# BIGTOP-3883:
# yum-utils, yum-priorities and yum-config-manager are NOT available in openEuler 22.03 .......
#.....
```

Reply:
ok, use setup-env-centos.sh file and use dnf instead of yum for openeuler.
code commit : 684f5c4857ddc11cd0768a9fca2f6af38dd07e59